### PR TITLE
Refactored rpc docs

### PR DESCRIPTION
**Pull request by rohan-varma**
Reorganize as per @jlin27 's comments. Screenshots added in comments.

### CODE REVIEW

**Comment by rohan-varma** [comment]
toctree

@jlin27 This is the RPC doc page with the `toctree` approach

File: docs/source/rpc/index.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+The distributed RPC framework provides mechanisms for multi-machine model training through a set of primitives to allow for remote communication, and a higher-level API to automatically differentiate
+models split across several machines.
+
+.. toctree::
+  :maxdepth: 1
+  
+  rpc.rst
+  distributed_autograd.rst
+  rref.rst
```
Comment (mrshenli) [inline docs/source/rpc/index.rst]: @jlin27 
Do we also need to add tutorials links here? I noticed that distributed tutorials are no longer available in the left navigation bar or tutorial landing page. Is that intentional?

**Comment by rohan-varma** [comment]
@jlin27 Most recent update:
refactor

Echoing @mrshenli , should we add tutorials here as well?

**Comment by mrshenli** [comment]
Shall we add a note on the top of the `rpc.rst` page saying that 

> APIs in the RPC package are stable. There are multiple ongoing work items to improve performance and error handling, which will ship in future releases.

**Comment by mrshenli** [comment]
> Echoing @mrshenli , should we add tutorials here as well?

Will need @jlin27's input. If we decided to remove PTD tutorial from the left navigation bar. We probably need to add them here.

**Comment by rohan-varma** [comment]
rpc_warning
@mrshenli

**Comment by jlin27** [comment]
The new rpc/index.rst landing page looks.
The PDT Tutorials will be moved over to a new recipes landing page because they fit that format better - succinct actionable explanations as opposed the longer format tutorials. I'm in the process of moving some of the exiting tutorials over the recipes landing page and will let you know when they are out so the can be linked from this landing page if desired.

The tutorials should still be accessible by their old urls if you would like to use those as placeholders for now. e.g. https://pytorch.org/tutorials/intermediate/ddp_tutorial.html

**Review by jlin27 — CHANGES_REQUESTED** [review]
New build looks good. I added my comments about adding tutorials/recipes in the main conversation.

**Comment by rohan-varma** [comment]
doc_updat_rpc

@mrshenli @jlin27 , this is the latest update

**Review by jlin27 — APPROVED** [review]

**Pull request closed without merging**